eng/devices/android.cake

```diff
@@ -213,8 +213,15 @@ Task("Test")
 	Information("Test App Package Name: {0}", TEST_APP_PACKAGE_NAME);
 	Information("Test App Instrumentation: {0}", TEST_APP_INSTRUMENTATION);
 	Information("Test Results Directory: {0}", TEST_RESULTS);
-
-	CleanDirectories(TEST_RESULTS);
+	
+	if (!IsCIBuild())
+		CleanDirectories(TEST_RESULTS);
+	else
+	{
+		// Because we retry on CI we don't want to delete the previous failures
+		// We want to publish those files for reference
+		DeleteFiles(Directory(TEST_RESULTS).Path.Combine("*.*").FullPath);
```

Comment:
Is this ok @rmarinho? Will there be results left over from other CI runs that we need to clear out?

Reply:
Hum, the macos images are not clear, so in a next test run there could be files there?

I wonder if we just need a step to clear this on the yaml timeline when the pipeline starts